azext_iot/tests/digitaltwins/test_dt_model_unit.py

```diff
@@ -131,32 +133,44 @@ def test_add_models(
         assert request_body == expected_payload
         assert result == json.loads('[' + generic_result + ']')
 
-    @pytest.mark.usefixtures("set_cwd")
-    @responses.activate
-    def test_large_ontology_error(self, fixture_cmd, fixture_dt_client):
-        models_added = []
-        models_deleted = []
-
-        def post_request_callback(request):
+    def configure_large_ontology_create_response(self, responses, models_added):
+        def post_request_callback_first(request):
             payload = json.loads(request.body)
+            # Check the batch order starts from model with least dependency
```

Comment:
This feedback item is more about, are we losing test accuracy by the refactor to a less specific assertion

Reply:
Hi Paymaun, the reason I change the original check to this is due to: I found first model been added to the list is different from windows vs. ubuntu/macOS, for windows it's **dtmi:digitaltwins:rec_3_3:core:Agent;1**, however for ubuntu/macOS it's **dtmi:digitaltwins:rec_3_3:business:Role;1**, they **both** have 0 model dependency, my assumption is the directory is scanned differently windows vs. ubuntu/macOS. The reason why we didn't got test failure is due to the assertion code under CliErrors was _never executed_ due to wrong indentation before this pr. Therefore i need to change a way to test this dtmi:digitaltwins:rec_3_3:core:Agent;1, since it's not always the first entry. Or I can change the assertion into `assert model_ids[0] == "dtmi:digitaltwins:rec_3_3:core:Agent;1" or model_ids[0] == "dtmi:digitaltwins:rec_3_3:business:Role;1"`, what do you think?😀

Reply:
Think of another way to make the assertion specific without checking the hardcoded id: I can get the dependency length of first entry directly from the model definition and make sure it's 0, just updated the code, please let me know if this work, thanks!